atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+        for n in range(n_atoms):
+            sec_atom = sec_method.m_create(AtomParameters)
+            for key in self.atom_parameters.keys():
+                setattr(sec_atom, key, self.atom_parameters[key][n])
+
+        # Get the interactions
+        connectivity = self.h5md_groups['connectivity']
+        if not connectivity:
+            return
+
+        atom_labels = self.atom_parameters.get('label')
+        interaction_keys = ['bonds', 'angles', 'dihedrals', 'impropers']
+        interactions_by_type = []
+        for interaction_key in interaction_keys:
+            interaction_list = self._data_parser.hdf5_getter(connectivity, interaction_key)
+            if interaction_list is None:
+                continue
+            elif type(interaction_list) == h5py.Group:
+                self.logger.warning('Time-dependent ' + key + ' currently not supported. ' + key + ' list will not be stored')
+                continue
+
+            interaction_type_dict = {
+                'type': interaction_key,
+                'n_interactions': len(interaction_list),
+                'n_atoms': len(interaction_list[0]),
+                'atom_indices': interaction_list,
+                'atom_labels': np.array(atom_labels)[interaction_list] if atom_labels is not None else None
+            }
+            interactions_by_type.append(interaction_type_dict)
+        self.parse_interactions_by_type(interactions_by_type, sec_model)
+
+        # Get the force calculation parameters
+        force_calculation_parameters = self.parameter_info.get('force_calculations')
+        if force_calculation_parameters is None:
+            return
+
+        sec_force_calculations = sec_force_field.m_create(ForceCalculations)
+        sec_neighbor_searching = sec_force_calculations.m_create(NeighborSearching)
+
+        for key, val in force_calculation_parameters.items():
+            if not isinstance(val, dict):
+                key = self.check_metainfo_for_key_and_Enum(ForceCalculations, key, val)
+                setattr(sec_force_calculations, key, val)
+            else:
+                if key == 'neighbor_searching':
+                    for neigh_key, neigh_val in val.items():
+                        neigh_key = self.check_metainfo_for_key_and_Enum(NeighborSearching, neigh_key, neigh_val)
+                        setattr(sec_neighbor_searching, neigh_key, neigh_val)
+                else:
+                    self.logger.warning(key + 'is not a valid force calculations section. Corresponding parameters will not be stored.')
+
+    def parse_workflow(self):
+
+        workflow_parameters = self.parameter_info.get('workflow').get('molecular_dynamics')
+        # TODO should store parameters that do not match the enum vals as x_h5MD params, not sure how with MDParser??
+        if workflow_parameters is None:
+            return
+
+        def get_workflow_results(property_type_dict, observables, workflow_results):
```

Comment:
i suggest you make this a class method.. do not pass workflow_results as argument, simply return property_dict and update workflow_results outside the function.

Reply:
be verbose with the arguments:
```python
def get_workflow_property(observables, property_key=None, property_value_key=None....)
```
then you call it
get_workflow_property(ensemble_average_observables, **ensemble_property_dict)

Reply:
I think I at least mostly executed what you wanted here. You probably should recheck in the next review round.